### ISSUE TITLE
(info) Is 8822cu also supported?

### ISSUE THREAD

**Issue by SunboX** [morrownr/8821cu-20210916#79]
Will this driver also work with the 8822cu chipset?

**Comment by morrownr** [comment]
No. You are the first person I have had contact with that has an adapter with that chipset. I would need source code and an adapter to add support. It may not be easy to find the source. The retailer that I am usually able to find source does not carry any adapters with that chipset so does not have the source available.

Maybe I can look into supporting this chipset but I may need you to help track down the source. There is also the in-kernel driver for this chipset. As of kernel 6.2, there is an in-kernel driver. The performance is not that good yet.

If it were me, and you are still able to return it for exchange or money back, that is what I would do. The best chipsets, if you have no alternative but Realtek, is the rtl8812bu and rtl8811cu chipsets. If you have the option, get an adapter with a mt7921au or mt7612u or mt7610u chipset as those chipsets use in-kernel drivers.

Look for menu item 2 at the following link for a good list of adapters based on the above chipsets:

https://github.com/morrownr/USB-WiFi

**Comment by birdofprey** [comment]
os: debian Gnu/Linux 12
kernel: Linux localhost 6.4.0-0.deb12.2-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.4.4-3~bpo12+1 (2023-08-08) x86_64 GNU/Linux
dkms status: rtl8821cu/5.12.0.4, 6.4.0-0.deb12.2-amd64, x86_64: installed
installed okay!!! butiful

**Comment by waddisdis** [comment]
> os: debian Gnu/Linux 12
> kernel: Linux localhost 6.4.0-0.deb12.2-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.4.4-3~bpo12+1 (2023-08-08) x86_64 GNU/Linux
> dkms status: rtl8821cu/5.12.0.4, 6.4.0-0.deb12.2-amd64, x86_64: installed
> installed okay!!! butiful

You got 8822cu with this driver to work with both antennas?

**Comment by waddisdis** [comment]
> os: debian Gnu/Linux 12
> kernel: Linux localhost 6.4.0-0.deb12.2-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.4.4-3~bpo12+1 (2023-08-08) x86_64 GNU/Linux
> dkms status: rtl8821cu/5.12.0.4, 6.4.0-0.deb12.2-amd64, x86_64: installed
> installed okay!!! butiful

@birdofprey 
You got 8822cu with this driver to work with both antennas?

**Comment by ADV-LUT** [comment]
Hello @morrownr ,

I am facing the same issue and am looking for a driver for the RTL8822cu chipset.
My kernel: Linux 5.15.136-tegra (Ubuntu 22.04)

Is there a driver available yet?

**Comment by morrownr** [comment]
Hi @ADV-LUT 

I suggest you test the rtw88 driver as outlined in the first message at this link:

https://github.com/morrownr/88x2bu-20210702/issues/224

We need reports from uses of the 8822cu. Please pay attention to usb3 and usb2 support and if you need instructions, let me know.
